apypie/action.py

```diff
@@ -10,49 +10,66 @@
 except NameError:  # Python 3 has no basestring
     basestring = str  # pylint: disable=W0622
 
+try:
+    from typing import Optional, Any, List, TYPE_CHECKING
+except ImportError:
+    TYPE_CHECKING = False
+
+if TYPE_CHECKING:
+    from apypie.api import Api
+
 
 class Action:
     """
     Apipie Action
     """
 
     def __init__(self, name, resource, api):
+        # type: (str, str, Api) -> None
         self.name = name
         self.resource = resource
         self.api = api
 
     @property
     def apidoc(self):
+        # type: () -> dict
```

Comment:
Should this be [Dict](https://docs.python.org/3/library/typing.html#typing.Dict) with specific typing about the key and value?

Reply:
That would imply to document the contents of the `apidoc.json` here, which feels the wrong place?

Reply:
Maybe. I'm not sure of when to use `dict` vs `Dict` when you don't specify the exact type.

Reply:
My understanding was that it doesn't matter if you don't specify the contents (and don't need things like Iterable or Sequence, which you shouldn't use for return values anyways).

Reply:
I since read https://www.python.org/dev/peps/pep-0585/ and using `typing.Dict` is deprecated in Python 3.9 where you can use `dict[str, int]` as well. I think that answers my question.